gammapy/modeling/models/core.py

```diff
@@ -274,7 +282,13 @@ def to_dict(self):
         for model in self._models:
             model_data = model.to_dict()
             models_data.append(model_data)
-        return {"components": models_data}
+        if self._covariance.filename is not None:
+            return {
+                "components": models_data,
+                "covariance": str(self._covariance.filename),
```

Comment:
I think here we can just rely on a default filename create from the models filename like `models.yaml` and `models_covariance.ecsv` or `my_model.yaml` and `my_model_covariance.ecsv`. More flexibility is then provided with the `Model.write_covariance()` and `Model.read_covariance()` API.

Reply:
I define the name on models.write() using the same prefix than for the models.yaml file, now it is passed  through self._covar_file

Reply:
OK, sounds good...